cocos/renderer/gfx-validator/CommandBufferValidator.cpp

```diff
@@ -0,0 +1,228 @@
+/****************************************************************************
+ Copyright (c) 2020-2021 Xiamen Yaji Software Co., Ltd.
+
+ http://www.cocos.com
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated engine source code (the "Software"), a limited,
+ worldwide, royalty-free, non-assignable, revocable and non-exclusive license
+ to use Cocos Creator solely to develop games on your target platforms. You shall
+ not use Cocos Creator software for developing other software or tools that's
+ used for developing games. You are not granted to publish, distribute,
+ sublicense, and/or sell copies of Cocos Creator.
+
+ The software or tools in this License Agreement are licensed, not sold.
+ Xiamen Yaji Software Co., Ltd. reserves all rights not expressly granted to you.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ THE SOFTWARE.
+****************************************************************************/
+
+#include "base/CoreStd.h"
+#include "base/job-system/JobSystem.h"
+#include "base/threading/MessageQueue.h"
+
+#include "BufferValidator.h"
+#include "CommandBufferValidator.h"
+#include "DescriptorSetValidator.h"
+#include "DeviceValidator.h"
+#include "FramebufferValidator.h"
+#include "InputAssemblerValidator.h"
+#include "PipelineStateValidator.h"
+#include "QueueValidator.h"
+#include "RenderPassValidator.h"
+#include "TextureValidator.h"
+
+namespace cc {
+namespace gfx {
+
+CommandBufferValidator::~CommandBufferValidator() {
+    CC_SAFE_DELETE(_actor);
+}
+
+void CommandBufferValidator::doInit(const CommandBufferInfo &info) {
+    CommandBufferInfo actorInfo = info;
+    actorInfo.queue             = ((QueueValidator *)info.queue)->getActor();
+
+    _actor->initialize(actorInfo);
+}
+
+void CommandBufferValidator::doDestroy() {
+    _actor->destroy();
+}
+
+void CommandBufferValidator::begin(RenderPass *renderPass, uint subpass, Framebuffer *framebuffer) {
+    CCASSERT(!_insideRenderPass, "Already inside an render pass?");
+
+    _insideRenderPass = !!renderPass;
```

Comment:
It is not clear to understand the logic.

Reply:
Added some comments to that